File: code-examples/php/indexes/multikey.php
<?php

require_once __DIR__ . "/vendor/autoload.php";

// Replace the placeholders with your credentials
$uri = 'mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority';

$client = new MongoDB\Client($uri);

try {
    $database = $client->selectDatabase('sample_mflix');
    $movies = $database->selectCollection('movies');

    // Create a multikey index on the "cast" field in the "movies" collection
    $result = $movies->createIndex(['cast' => 1]);

    // Print the result of creating the index
    echo "Index created: $result\n";

    // Query to find movies cast by "Viola Davis"
    $query = ['cast' => 'Viola Davis'];
    $projection = ['_id' => 0, 'cast' => 1, 'title' => 1];
    $options = ['projection' => $projection];

    $cursor = $movies->find($query, $options);

    foreach ($cursor as $document) {
        var_dump($document);
    }

} catch(MongoDB\Driver\Exception\Exception $e) {
    echo $e->getMessage(), "\n";
    echo $e->getFile(), "\n";
    echo $e->getLine(), "\n";
    exit;
} finally {
    $client = null;
}
?>

File: code-examples/scala/indexes/multikey.scala
import org.mongodb.scala._
import scala.concurrent.ExecutionContext.Implicits.global
import scala.util.{Failure, Success}

// Replace the placeholders with your credentials
val mongoClient: MongoClient = MongoClient("mongodb://<user>:<password>@<cluster-url>?writeConcern=majority")

val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
val collection: MongoCollection[Document] = database.getCollection("movies")

// Create a multikey index on the "cast" field in the "movies" collection
val indexResult = collection.createIndex(Document("cast" -> 1)).head()

indexResult.onComplete {
  case Success(indexName) => println(s"Index created: $indexName")
  case Failure(failure) => println(s"Failed: $failure")
}

// Prepare a query to find movies where "Viola Davis" is in the cast
val query: Document = Document("cast" -> "Viola Davis")

// Specify the fields we want to return
val projection: Document = Document("_id" -> 0, "cast" -> 1 , "title" -> 1)

val cursor: FindObservable[Document] = collection.find(query).projection(projection)

cursor.subscribe((doc: Document) => println(doc.toJson()))

// Close the MongoClient, this is scala so the client will not close automatically
mongoClient.close()
